src/maybe.ts

```diff
@@ -1596,58 +1596,36 @@ export function last<T>(array: Array<T | null | undefined>): Maybe<T> {
 }
 
 /**
-  Convert the arguments to a single `Maybe`. Useful for dealing with arrays of
-  `Maybe`s, via the spread operator.
-
+  Given an array or tuple of `Maybe`s, return a `Maybe` of the array or tuple
+  values.
+
+  -   Given an array of type `Array<Maybe<A> | Maybe<B>>`, the resulting type is
+      `Maybe<Array<A | B>>`.
+  -   Given a tuple of type `[Maybe<A>, Maybe<B>]`, the resulting type is
+      `Maybe<[A, B]>`.
+
+  If any of the items in the array or tuple are `Nothing`, the whole result is
+  `Nothing`. If all items in the array or tuple are `Just`, the whole result is
+  `Just`.
+      
   ## Examples
 
+  Given an array with a mix of `Maybe` types in it, both `allJust` and `mixed`
+  here will have the type `Maybe<Array<string | number>>`, but will be `Just`
+  and `Nothing` respectively.
+
   ```ts
   import Maybe from 'true-myth/maybe';
 
   let valid = [Maybe.just(2), Maybe.just('three')];
-  Maybe.all(...valid); // => Just([2, 'three']);
+  let allJust = Maybe.all(valid); // => Just([2, 'three']);
 
-  let invalid = [Maybe.just(2), Maybe.nothing<string>()];
-  Maybe.all(...invalid); // => Nothing
+  let invalid: Invalid = [Maybe.just(2), Maybe.nothing<string>()];
```

Comment:
Where is this `: Invalid` type coming from?

Reply:
My imagination, as far as I can tell. 😂 Thanks for catching it; I've removed it.